### PR TITLE
build(schema): Add '$schema' property

### PR DESCRIPTION
**Pull request by caoli5288**
#### Description

schema.json should allow $schema property

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by rashil2000** [comment]
I have thought about this previously. 

Pretty useful for autocompleting fields in Neovim, VSCode etc.

**Review by rashil2000 — APPROVED** [review]

**Comment by niheaven** [comment]
Why define a `$shema` property in JSON Schema? I mean, what is the purpose of this field in manifest?

**Comment by caoli5288** [comment]
> Why define a $shema property in JSON Schema? I mean, what is the purpose of this field in manifest?
@niheaven Will warn "$schema property not allow here" by VSCode if use this schema explicitly.

**Comment by niheaven** [comment]
Err, what is the situation of using this schema explicitly? In bucket's `.vscode` folder, this schema is defined to parse all `bucket/*.json`, and put `$schema` at the top of manifest json file is really strange, IMO.

**Comment by rashil2000** [comment]
> Err, what is the situation of using this schema explicitly? 

It can help in code autocompletion, and warn about unsupported fields right there in the text editor, which can save time. It's optional anyway, so won't break anything.

> In bucket's `.vscode` folder, this schema is defined to parse all `bucket/*.json`

Well, not everyone uses VSCode 😉

> put `$schema` at the top of manifest json file is really strange, IMO.

It is the standard practice. Look at Windows Terminal's settings schema, for example: https://raw.githubusercontent.com/microsoft/terminal/release-1.11/doc/cascadia/profiles.schema.json. When you install Windows Terminal for the first time, the settings file already contains these 2 fields: 

image

This helps in easy discoverability of supported features and fields.

**Comment by niheaven** [comment]
Setting config file that used `$schema` is acceptable, I thought, but for vast manifests... It's okay for adding it here for some fallback reasons, though I hope it will never be used 🤣 

Please modify the title and add changelog item for this PR.

**Comment by caoli5288** [comment]
> Setting config file that used `$schema` is acceptable, I thought, but for vast manifests... It's okay for adding it here for some fallback reasons, though I hope it will never be used 🤣
> 
> Please modify the title and add changelog item for this PR.

@niheaven I make this PR from master branch, so I can't modify the changelog. Maybe you can help me modify it?

**Comment by rashil2000** [comment]
Your PR is just 4 line change, you can close this and create new one from `develop` to `develop`.

**Comment by niheaven** [comment]
Aha, got it. But you need to revert the recent two commits after this PR is merged in your forked repo.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@
 - **decompress:** Fix nested Zstd archive extraction ([#4608](https://github.com/ScoopInstaller/Scoop/issues/4608))
 - **shim:** Fix PS1 shim error when in different drive in PS7 ([#4614](https://github.com/ScoopInstaller/Scoop/issues/4614))
 
+### Builds
```
Comment (rashil2000) [inline CHANGELOG.md:11]: Is `Builds` the appropriate category here? I'm not so sure.

**Review by rashil2000 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (niheaven) [reply CHANGELOG.md:11]: I put all schema and bin changes under `Builds` since this has less CLI changes to end users.

**Pull request merged**